src/psp/guest_request_cmd.rs

```diff
@@ -0,0 +1,809 @@
+        } else {
+            // Set decrypt operation
+            let ret: c_int = EVP_DecryptInit_ex(
+                ctx,
+                EVP_aes_256_gcm(),
+                ptr::null_mut(),
+                ptr::null(),
+                ptr::null(),
+            );
+            if ret != 1 {
+                prints!("ERR: EVP_DecryptInit_ex failed, rc={}\n", ret);
+                return Err(());
+            }
+        }
+
+        // Provide key size
+        let __vmpck_size: c_int = match c_int::try_from(VMPCK_SIZE) {
+            Ok(c) => c,
+            Err(_) => {
+                prints!("ERR: VMPCK too big for c_int\n");
+                return Err(());
+            }
+        };
+        let mut ret: c_int = EVP_CIPHER_CTX_set_key_length(ctx, __vmpck_size);
+        if ret != 1 {
+            prints!(
+                "ERR: Failed to set the key length for encryption ({})\n",
+                ret
+            );
+            return Err(());
+        }
+
+        // Provide iv size
+        let __gcm_set_ivlen: c_int = match c_int::try_from(EVP_CTRL_GCM_SET_IVLEN) {
+            Ok(c) => c,
+            Err(_) => {
+                prints!("ERR: Operation SET_IVLEN too big for c_int\n");
+                return Err(());
+            }
+        };
+        let __iv_size: c_int = match c_int::try_from(IV_SIZE) {
+            Ok(c) => c,
+            Err(_) => {
+                prints!("ERR: IV_SIZE too big for c_int\n");
+                return Err(());
+            }
+        };
+        ret = EVP_CIPHER_CTX_ctrl(ctx, __gcm_set_ivlen, __iv_size, ptr::null_mut());
+        if ret != 1 {
+            prints!("ERR: EVP_CIPHER_CTX_ctrl failed ({})\n", ret);
+            return Err(());
+        }
+
+        // Provide key and iv
+        let svsm_secrets_va: VirtAddr = get_svsm_secrets_page();
+        let svsm_secrets_ptr: *mut SnpSecrets = svsm_secrets_va.as_mut_ptr();
+        let key: [u8; VMPCK_SIZE] = (*svsm_secrets_ptr).vmpck0();
+
+        let mut iv: [u8; IV_SIZE] = [0u8; IV_SIZE];
+        iv[..U64_SIZE].copy_from_slice(&u64::to_ne_bytes(*seqno));
+
+        if encryption {
+            ret = EVP_EncryptInit_ex(
+                ctx,
+                ptr::null(),
+                ptr::null_mut(),
+                &key as *const _ as *const u8,
+                &iv as *const _ as *const u8,
+            );
+            if ret != 1 {
+                prints!("ERR: EVP_EncryptInit_ex failed ({})\n", ret);
+                return Err(());
+            }
+        } else {
+            ret = EVP_DecryptInit_ex(
+                ctx,
+                ptr::null(),
+                ptr::null_mut(),
+                &key as *const _ as *const u8,
+                &iv as *const _ as *const u8,
+            );
+            if ret != 1 {
+                prints!("ERR: EVP_DecryptInit_ex failed, rc={}\n", ret);
+                return Err(());
+            }
+        }
+
+        Ok(ctx)
+    }
+
+    unsafe fn encrypt_request_unsafe(
```

Comment:
No need for the `x_unsafe()`, we haven't done this before

Reply:
That was one of those moments that no good function name come to mind. :-)
I renamed it to `__encrypt_request()`, it might not be Rust idiomatic but hopefully better than `x_unsafe()`. 
I did the same for `decrypt_request_unsafe()`

Reply:
Perfect :) thanks!